### PR TITLE
Fix GUI language selection

### LINKED ISSUE #6090: Translations don't work when running through PyInstaller
Apparently `translator.load(directory=<path>)` can't find translation files when running with Pyinstaller.

### PR DESCRIPTION
**Pull request by ichorid**
fixes #6090 

First of all, the translation mechanism _never really worked_ for several reasons:
1. The translations dir (`i18n`) was not included in pyinstaller 
2. The translations dir path was hardcoded without accounting for the possibility of running under pyinstaller
3. The mechanism to detect system language was provided by Python `locale` module which is unreliable and not natively compatible with QT locales.
4. On some systems even QT can have hard times detecting the UI language, so it is necessary to provide some means for the user to **select** their preferred GUI language.

This PR fixes all of the above problems. Among the other things, it was necessary to move `tr()` wrapper to `utilities.py` to solve recursive imports, and move QT settings to be read *before* the actual GUI window starts.

In addition, this PR further improves and simplifies the Settings page by replacing label-checkbox pairs with pure labelled checkboxes, and removes flow-breaking "confirmation" dialog with a notification message.

image

### CODE REVIEW

**Review by xoriole — APPROVED** [review]

**Review by devos50 — COMMENTED** [review]
Would it be much trouble to add a GUI test for this?

File: src/tribler-gui/tribler_gui/utilities.py [hunk 1]
```diff
@@ -417,3 +441,23 @@ def dict_item_is_any_of(d, key, values):
     if not d or not key or not values:
         return False
     return key in d and d[key] in values
+
+
+def translate(context, key):
```
[the remark this answers is not in the document]
Comment (ichorid) [reply src/tribler-gui/tribler_gui/utilities.py:446]: :facepalm:

File: src/tribler-gui/tribler_gui/widgets/settingspage.py [hunk 2]
```diff
@@ -63,6 +63,16 @@ def initialize_settings_page(self):
...
+        self.window().language_selector.addItems(self.lang_list)
+
+        selected_lang_code = self.window().gui_settings.value('translation', None)
+        if selected_lang_code is not None:
+            for lang_name, lang_code in AVAILABLE_TRANSLATIONS.items():
+                if selected_lang_code == lang_code:
+                    self.window().language_selector.setCurrentIndex(self.lang_list.index(lang_name))
+                    break
+
+        # connect(self.window().language_selector.currentIndexChanged, self.on_language_selector_changed)
```
[the remark this answers is not in the document]
Comment (ichorid) [reply src/tribler-gui/tribler_gui/widgets/settingspage.py:75]: :facepalm:

File: src/tribler-gui/tribler_gui/utilities.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ichorid) [reply src/tribler-gui/tribler_gui/utilities.py:446]: I'll address this stuff in the next PR silently...

**Pull request merged**